sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -97,7 +98,7 @@ private[sql] class SharedState(
   val statusStore: SQLAppStatusStore = {
     val kvStore = sparkContext.statusStore.store.asInstanceOf[ElementTrackingStore]
     val listener = new SQLAppStatusListener(conf, kvStore, live = true)
-    sparkContext.listenerBus.addToStatusQueue(listener)
+    sparkContext.listenerBus.addToQueueIfNotExists(listener, LiveListenerBus.APP_STATUS_QUEUE)
```

Comment:
why are we having to make APP_STATUS_QUEUE visible versus make addToStatusQueue have a parameter?